### PR TITLE
fix(cb2-8705): fix to observables & higher and lower values

### PR DESCRIPTION
**Pull request by Daniel-Searle**
##   Fix to observables & higher and lower values

[CB2-8705](https://dvsa.atlassian.net/browse/CB2-8705)

## Checklist

- [x] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [x] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/components/approval-type/approval-type.component.ts [hunk 1]
```diff
@@ -142,25 +151,44 @@ export class ApprovalTypeInputComponent extends BaseControlComponent implements
...
   }
+
   private extractValuesNtaIva(value: string, pattern: RegExp) {
-    const matches = value.match(pattern);
+    const matches = value.match(pattern)?.map(x => (x.length > 25 ? x.substring(0, 25) : x));
     this.setTypeApprovalNumbers(matches ?? []);
   }
 
   private setTypeApprovalNumbers(matches: string[]) {
+    console.log('setting approval numbers');
```
Comment (gjulien-bjss) [inline src/app/forms/components/approval-type/approval-type.component.ts:189]:
```suggestion
```

File: src/app/forms/components/approval-type/approval-type.component.ts [hunk 2]
```diff
@@ -142,25 +151,44 @@ export class ApprovalTypeInputComponent extends BaseControlComponent implements
...
     if (matches) {
+      console.log('matches');
```
Comment (gjulien-bjss) [inline src/app/forms/components/approval-type/approval-type.component.ts:191]:
```suggestion
```

File: src/app/forms/templates/general/approval-type.template.ts [hunk 3]
```diff
@@ -29,17 +29,18 @@ export const HgvAndTrlTypeApprovalTemplate: FormNode = {
...
               'NTA',
               'ECTA',
+              'ECSSTA',
               'IVA',
               'NSSTA',
               'GB WVTA',
               'UKNI WVTA',
               'EU WVTA Pre 23',
-              'EU WVTA 23',
+              'EU WVTA 23 on',
```
Comment (kcarr-bjss) [inline src/app/forms/templates/general/approval-type.template.ts:38]: on?

**Review by kcarr-bjss — DISMISSED** [review]

File: src/app/forms/templates/general/approval-type.template.ts (hunk 3 above)
Comment (Daniel-Searle) [reply src/app/forms/templates/general/approval-type.template.ts:38]: Approval type on ticket

**Review by gjulien-bjss — APPROVED** [review]

File: src/app/forms/components/approval-type/approval-type.component.ts [hunk 4]
```diff
@@ -179,103 +205,80 @@ export class ApprovalTypeInputComponent extends BaseControlComponent implements
...
-            errors: [
-              {
-                error: true,
-                reason: 'Approval type number is required with Approval type',
-                index: 0
-              }
-            ]
-          };
+        if (oneRequired()) {
+          console.log('setting errors 1');
```
Comment (gjulien-bjss) [inline src/app/forms/components/approval-type/approval-type.component.ts:246]:
```suggestion
```

File: src/app/forms/components/approval-type/approval-type.component.ts [hunk 5]
```diff
@@ -179,103 +205,80 @@ export class ApprovalTypeInputComponent extends BaseControlComponent implements
...
-              {
-                error: true,
-                reason: 'Approval type number is required with Approval type',
-                index: 0
-              }
-            ]
-          };
+      case 'UKNI WVTA':
+        if (fourRequired()) {
+          console.log('setting errors 4');
```
Comment (gjulien-bjss) [inline src/app/forms/components/approval-type/approval-type.component.ts:260]:
```suggestion
```

File: src/app/forms/components/approval-type/approval-type.component.ts [hunk 6]
```diff
@@ -179,103 +205,80 @@ export class ApprovalTypeInputComponent extends BaseControlComponent implements
...
-              {
-                error: true,
-                reason: 'Approval type number is required with Approval type',
-                index: 0
-              }
-            ]
-          };
+      case 'Small series':
+        if (threeRequired()) {
+          console.log(' setting errors 3');
```
Comment (gjulien-bjss) [inline src/app/forms/components/approval-type/approval-type.component.ts:268]:
```suggestion
```

**Review by kcarr-bjss — APPROVED** [review]

**Pull request merged**